```rust
use std::thread;
use std::time::Duration;

fn main() {
    thread::spawn(|| {
        for i in 1..10 {
            println!("Count in thread: {i}!");
            thread::sleep(Duration::from_millis(5));
        }
    });

    for i in 1..5 {
        println!("Main thread: {i}");
        thread::sleep(Duration::from_millis(5));
    }
}
```
- 线程均为守护程序线程，主线程不会等待这些线程。
- 线程紧急警报 (panic) 是彼此独立的。
    - 紧急警报可以携带载荷，并可以使用 `downcast_ref` 对载荷进行解压缩。

## 范围线程

常规线程不能从它们所处的环境中借用：
```rust
use std::thread;

fn foo() {
    let s = String::from("Hello");
    thread::spawn(|| {
        println!("Length: {}", s.len());
    });
}

fn main() {
    foo();
}
```
可以使用[范围线程](https://doc.rust-lang.org/std/thread/fn.scope.html)来实现此目的：
```rust
use std::thread;

fn main() {
    let s = String::from("Hello");

    thread::scope(|scope| {
        scope.spawn(|| {
            println!("Length: {}", s.len());
        });
    });
}
```
- 其原因在于，在 `thread::scope` 函数完成后，可保证所有线程都已联结在一起，使得线程能够返回借用的数据。
- 此时须遵守常规 Rust 借用规则：你可以通过一个线程以可变的方式借用，也可以通过任意数量的线程以不可变的方式借用。